fmriprep/interfaces/utils.py

```diff
@@ -87,49 +85,49 @@ def _run_interface(self, runtime):
         return runtime
 
 
-class CombineROIsInputSpec(BaseInterfaceInputSpec):
+class AddTPMsInputSpec(BaseInterfaceInputSpec):
     in_files = InputMultiPath(File(exists=True), mandatory=True, desc='input list of ROIs')
-    ref_header = File(exists=True, mandatory=True,
-                      desc='reference NIfTI file with desired output header/affine')
+    indexes = traits.List(traits.Int, desc='select specific maps')
 
 
-class CombineROIsOutputSpec(TraitedSpec):
+class AddTPMsOutputSpec(TraitedSpec):
     out_file = File(exists=True, desc='union of binarized input files')
 
 
-class CombineROIs(SimpleInterface):
+class AddTPMs(SimpleInterface):
     """Generate the union (logical or) of a series of ROI masks"""
-    input_spec = CombineROIsInputSpec
-    output_spec = CombineROIsOutputSpec
+    input_spec = AddTPMsInputSpec
+    output_spec = AddTPMsOutputSpec
 
     def _run_interface(self, runtime):
-        self._results['out_file'] = _combine_rois(self.inputs.in_files, self.inputs.ref_header)
-        return runtime
+        in_files = self.inputs.in_files
 
+        indexes = list(range(len(in_files)))
+        if isdefined(self.inputs.indexes):
+            indexes = self.inputs.indexes
 
-class ConcatROIsInputSpec(BaseInterfaceInputSpec):
-    in_file = File(exists=True, mandatory=True, desc='input file')
-    in_mask = File(exists=True, mandatory=True, desc='input mask')
-    ref_header = File(exists=True, mandatory=True,
-                      desc='reference NIfTI file with desired output header/affine')
+        if len(self.inputs.in_files) < 2:
+            self._results['out_file'] = in_files[0]
+            return runtime
 
+        first_fname = in_files[indexes[0]]
+        if len(indexes) == 1:
+            self._results['out_file'] = first_fname
+            return runtime
 
-class ConcatROIsOutputSpec(TraitedSpec):
-    out_file = File(exists=True, desc='output average file')
+        im = nb.load(first_fname)
+        data = im.get_data()
 
+        for idx in indexes[1:]:
+            data += nb.load(in_files[idx]).get_data()
 
-class ConcatROIs(SimpleInterface):
-    """Concatenate two ROI files along time axis
+        data[data > 1] = 1.0
+        data[data < 0] = 0.0
 
-    ``in_file`` is resampled to match ``in_mask``, and the two concatenated
-    datasets are saved with the affine and header of ``ref_header``.
-    """
-    input_spec = ConcatROIsInputSpec
-    output_spec = ConcatROIsOutputSpec
+        out_file = fname_presuffix(first_fname, suffix='_tpmsum')
+        im.__class__(data, im.affine, im.header).to_filename(out_file)
```

Comment:
I believe this is boolean? Might be worth making sure that it's saved as `np.uint8`.

Reply:
Ok, I'm double checking. That should be a float. This interface works directly on the TPMs, and it should not change the original dtype. I think it is safe to leave it this way.